pkg/e2e/verify/user_webhook.go

```diff
@@ -27,15 +27,390 @@ var _ = ginkgo.Describe(userWebhookTestName, func() {
+			Expect(err).NotTo(HaveOccurred())
+		}, float64(viper.GetFloat64(config.Tests.PollingTimeout)))
+
+		ginkgo.It("system:admin can manage customer users with other IDP and no group", func() {
+			userName := util.RandomStr(5) + "@customdomain"
+			identities := []string{"otherIDP:testing_string"}
+			h.Impersonate(rest.ImpersonationConfig{
+				UserName: "system:admin",
+				Groups: []string{
+					"system:authenticated",
+					"system:authenticated:oauth",
+				},
+			})
+			user, err := createUser(userName, identities, []string{}, h)
+			defer func() {
+				h.Impersonate(rest.ImpersonationConfig{})
+				deleteUser(user.Name, h)
+			}()
+			Expect(err).NotTo(HaveOccurred())
+			err = deleteUser(user.Name, h)
+			Expect(err).NotTo(HaveOccurred())
+		}, float64(viper.GetFloat64(config.Tests.PollingTimeout)))
+
+		// not testing osd-sre-admins because RBAC prevents them from managing
+		// users, even though webhook allows them to
+
+		// not testing system:serviceaccount:openshift-authentication:oauth-openshift
+		// because RBAC prevents it from managing users, even though webhook
+		// allows it to
+
+		// osd-sre-cluster-admins can manage protected RH users as long as the
+		// user is in one of the protected groups and is using the SRE IdP
+		ginkgo.It("osd-sre-cluster-admins can manage protected redhat users with SRE IDP and RH group", func() {
+			userName := util.RandomStr(5) + "@redhat.com"
+			identities := []string{"OpenShift_SRE:" + util.RandomStr(5)}
+			groups := []string{"osd-devaccess"}
+			// we need to add the username to the group before we create the user,
+			// because the user object cannot be created until the username is in
+			// the group
+			addUserToGroup(userName, groups[0], h)
+			h.Impersonate(rest.ImpersonationConfig{
+				UserName: "test-user@redhat.com",
+				Groups: []string{
+					"osd-sre-cluster-admins",
+					"system:authenticated",
+					"system:authenticated:oauth",
+				},
+			})
+			user, err := createUser(userName, identities, groups, h)
+			defer func() {
+				h.Impersonate(rest.ImpersonationConfig{})
+				deleteUser(user.Name, h)
+			}()
+			Expect(err).NotTo(HaveOccurred())
+			err = deleteUser(user.Name, h)
+			Expect(err).NotTo(HaveOccurred())
+		}, float64(viper.GetFloat64(config.Tests.PollingTimeout)))
+
+		// osd-sre-cluster-admins cannot create/update protected RH users if
+		// the user is using the SRE IdP but is not in one of the protected
+		// groups
+		ginkgo.It("osd-sre-cluster-admins cannot create/update protected redhat users with SRE IDP and no group", func() {
+			userName := util.RandomStr(5) + "@redhat.com"
+			identities := []string{"OpenShift_SRE:" + util.RandomStr(5)}
+			h.Impersonate(rest.ImpersonationConfig{
+				UserName: "test-user@redhat.com",
+				Groups: []string{
+					"osd-sre-cluster-admins",
+					"system:authenticated",
+					"system:authenticated:oauth",
+				},
+			})
+			user, err := createUser(userName, identities, []string{}, h)
```

Comment:
Message also mentions update as well. Thoughts on adding it into this test?

Reply:
Also, can osd-sre-cluster-admins delete these users? Just not create or update?

Reply:
Ah I see now, delete is further down

Reply:
I didn't add any updates to the test for simplicity. the code is the same, regardless of whether the user is trying to create or update a user (it only differs in the case of deleting the user), and testing update in addition to create adds a level of complexity to the tests (would need to create a `updateUser` function and attempt to determine _what_ to update in the user object that wouldn't cause unnecessary testing weirdness. considering the code is the same, I don't think the overhead is worth it

Reply:
Yep, that's totally fair. Maybe leave a comment to the effect of "only testing create for simplicity"?